pymc3/distributions/multivariate.py

```diff
@@ -1902,39 +1886,39 @@ def _quaddist(self, value):
         else:
             onedim = False
 
-        delta = value - self.mu
-        if self._isEVD:
-            sqrt_quad = kron_dot(self.QTs, delta.T)
-            sqrt_quad = sqrt_quad / at.sqrt(self.eigs[:, None])
-            logdet = at.sum(at.log(self.eigs))
+        delta = value - mu
+
+        # Setup EVD if applicable
+        if eigh_iterable.data.shape != (0,):
+            eigs_sep, Qs = zip(*eigh_iterable.data)  # Unzip
```

Comment:
Again a few hacks were required over here (using `.shape` for checking an empty `TensorVariable` and `.data` for accessing the data inside)

Reply:
Can this be more easily computed in `.dist`?